lib/beaker-hostgenerator/data.rb

```diff
@@ -164,6 +164,18 @@ def osinfo
             'template' => 'amazon-7-arm64'
           }
         },
+        'archlinuxrolling-64' => {
```

Comment:
Do we actually need rolling in here? I think platform needs it, but names don't.

Reply:
tested and updated it

Reply:
so the thing is, if the metadata.json only lists `"operatingsystem": "Archlinux"`, metadata2gha currently ignores Archlinux. with this:

```diff
diff --git a/metadata.json b/metadata.json
index 65a0664..0d70cff 100644
--- a/metadata.json
+++ b/metadata.json
@@ -38,7 +38,10 @@
   ],
   "operatingsystem_support": [
     {
-      "operatingsystem": "Archlinux"
+      "operatingsystem": "Archlinux",
+      "operatingsystemrelease": [
+        "rolling"
+      ]
     },
     {
       "operatingsystem": "Gentoo"
```

metadata2ghe recognizes it but renders it like this:
```
::set-output name=beaker_setfiles::[{"name":"Archlinux rolling","value":"archlinuxrolling-64{hostname=archlinuxrolling-64.example.com}"},...
```

Reply:
question is, is operatingsystemrelease required or not. I think none of the metadata.json that list Gentoo or Archlinux have that set.

Reply:
what I noticed:
rspec does ignore arch linux if operatingsystemrelease is listed in the metadata.json. That might be related to the older fact set https://github.com/camptocamp/facterdb/blob/7b822de15183da42b8b8fbaf7ef789c7b24a03cd/facts/3.14/archlinux-x86_64.facts#L392-L405 that were generated while lsb-release packages were not a dependency of facter.

Reply:
I [updated facterdb](https://github.com/camptocamp/facterdb/pull/162) but rspec-puppet-facts still ignores Arch Linux.

Reply:
Perhaps we should have both in beaker-hostgenerator. Alias `archlinux-64` to `archlinuxrolling-64`. I also wouldn't be opposed to special casing a missing `operatingsystemrelease` entry in puppet_metadata's beaker support. In https://github.com/ekohl/puppet_metadata/blob/4e63a22f7c114f0ee5edde93657f9d2fa3ae8fde/lib/puppet_metadata/metadata.rb#L77-L92 we already interpret it as "supports all versions" (note the `releases.nil?` part).

What do you think about that?

Reply:
The easiest way to alias is to modify `osinfo` at the end:
https://github.com/voxpupuli/beaker-hostgenerator/blob/2adeb8111ce78f12fdf87d957c66740df427fd12/lib/beaker-hostgenerator/data.rb#L1745-L1748

There you can add `result['archlinux-64'] = result['archlinuxrolling-64']`.